pyNN/common/populations.py

```diff
@@ -828,6 +832,7 @@ def __init__(self, parent, selector, label=None):
         self.local_cells = self.all_cells[self._mask_local]
         self.first_id = numpy.min(self.all_cells)  # only works if we assume all_cells is sorted, otherwise could use min()
         self.last_id = numpy.max(self.all_cells)
+        self.annotations = {}
```

Comment:
I can't now remember if there was a good reason for `PopulationView` not having annotations, but in any case I don't think it is related to the recording issue, so should probably be removed from this PR and discussed in a separate PR.